lib/pbench/cli/agent/commands/results/__init__.py

```diff
@@ -1,94 +1,199 @@
-    logger = get_pbench_logger("pbench", config)
 
-    controller = os.environ.get("_pbench_full_hostname")
-    if not controller:
-        logger.error("Missing controller name (should be 'hostname -f' value)")
-        sys.exit(1)
+class MoveResults(BaseCommand):
+    def __init__(self, context):
+        super().__init__(context)
 
-    results_webserver = config.results.get("webserver")
-    if not results_webserver:
-        logger.error(
-            "No web server host configured from which we can fetch the FQDN of the host to which we copy/move results"
-        )
-        logger.debug("'webserver' variable in 'results' section not set")
+    def execute(self):
+        config = PbenchAgentConfig(self.context.config)
+        logger = get_pbench_logger("pbench-agent", config)
 
-    server_rest_url = config.results.get("server_rest_url")
-    response = requests.get(f"{server_rest_url}/host_info")
-    if response.status_code not in [200, 201]:
-        logger.error(
-            "Unable to determine results host info from %s/host_info", server_rest_url
-        )
-        sys.exit(1)
-    if response.text.startswith("MESSAGE"):
-        message = response.text.split("===")[1]
-        logger.info("*** Message from sysadmins of %s:", results_webserver)
-        logger.info("***\n*** %s", message)
-        logger.info("***\n*** No local actions taken.")
-        sys.exit(1)
-    results_path_prefix = response.text.split(":")[1]
-    if not results_path_prefix:
-        logger.error(
-            "fetch results host info did not contain a path prefix: %s", response.text
-        )
-        sys.exit(1)
+        results_webserver = config.get("results", "webserver")
+        if not results_webserver:
+            logger.error(
+                "No web server host configured from which we can fetch the FQDN of the host to which we copy/move results"
+            )
+
+        # FIXME: Pytest Failure
+        #
+        # server_rest_url = config.get("results", "server_rest_url")
+        # response = requests.get(f"{server_rest_url}/host_info")
+        # if not response.ok:
+        #     logger.error(
+        #         "Unable to determine results host info from %s/host_info",
+        #         server_rest_url,
+        #     )
+        #     sys.exit(1)
+        # if response.text.startswith("MESSAGE"):
+        #     message = response.text.split("===")[1]
+        #     print(f"*** Message from sysadmins of %s:", results_webserver, file=sys.stderr)
+        #     print(f"***\n*** %s", message, file=sys.stderr)
+        #     print(f"***\n*** No local actions taken.", file=sys.stderr)
+        #     sys.exit(1)
+        # results_path_prefix = response.text.split(":")[1]
+        # if not results_path_prefix:
+        #     logger.error(
+        #         "Fetch results host info did not contain a path prefix: %s",
+        #         response.text,
+        #     )
+        #     sys.exit(1)
```

Comment:
@dbutenhof, @portante:  can you guys confirm whether we need to retain this commented-out code?

It makes a `GET` request to the `/host_info` endpoint:  if the request fails or returns a `MESSAGE`, then the tool exits; otherwise, it fetches a "path prefix" from the response which it then, to the best of my knowledge, proceeds to ignore.

Dave mentioned elsewhere that this exchange should be rolled up into the `PUT` request, but it doesn't seem like it matters much.  Are you guys good with me deleting this code from here?

Reply:
We don't.

The current SSH mechanism uses a 2 step scheme to determine exactly what the SSH host name is.  If first hits a URL to fetch that host name, and then pulls the host name from the response.  It checks to see if the system was put under maintenance mode first before trying to SSH to the host.

This level of indirection we use all the time when the server is under maintenance without having to turn off SSH.

In the new PUT API scheme, we can have Nginx, or what ever reverse proxy we place in front of it serve the same purpose: return a maintenance mode response instead.

Reply:
The intention is that `pbench-move-results` won't copy (in this case, `PUT`) the file if the server says it's not in operational mode. When you first install, it's not, for example. Right now, switching modes means re-linking files either manually or via an Ansible script.

We really want `PUT` to "know" the server mode and respond with an appropriate status code and message if it's not enabled... maybe initially by reading said file link, but eventually via internal data that's controlled by admin commands.

In any case, `pbench-move-results` won't be using this separate mechanism... which doesn't actually work the way it's suggested here, anyway. The `/host_info` just tells where to look.

Yeah, I'd say this can be deleted.

Reply:
Thank you!  (It's gone, now.)